.github/workflows/budibase_ci.yml

```diff
@@ -22,44 +22,51 @@ jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Use Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 14.x
+          cache: "yarn"
       - run: yarn
       - run: yarn lint
 
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           submodules: true
           token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
       - name: Use Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 14.x
+          cache: "yarn"
       - run: yarn
-      - run: yarn bootstrap
       - run: yarn build
 
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           submodules: true
           token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
       - name: Use Node.js 14.x
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 14.x
+          cache: "yarn"
       - run: yarn
-      - run: yarn bootstrap
-      - run: yarn build
+      - run: yarn build --scope=@budibase/types --scope=@budibase/shared-core --scope=@budibase/string-templates
+      - name: Get number of CPU cores
+        id: cpu-cores
+        uses: SimenB/github-actions-cpu-cores@3e824c4e1e0f0880c1e1023cbd7a418f5929d358
       - run: yarn test --ignore=@budibase/pro
+        env:
+          JEST_MAX_WORKERS: ${{ steps.cpu-cores.outputs.count }}
```

Comment:
The default behaviour for jest should be to use the cpu core available as max workers. We should omit the `max-workers` arg completely to enable this. 

We also have `scripts/test.sh` scripts in worker and server that always runs in band when in CI. this brought down time in CI significantly. This might conflict with the above. 

From past experiments, the server and worker tests perform very badly when using max available cores, given that so many of our tests start the whole service, it causes resource utilisation problems in CI.

Reply:
See you have updated the test scripts 👍 Other points remain

Reply:
That was an attempt to see any improvement, it did not work